locals.tf

```
@@ -22,5 +22,22 @@ locals {
     # default host in the resulting user object as well
     for user in var.users : "${user.name}@${user.host == null ? "%" : user.host}" => defaults(user, { host = "%" })
   }
-  postgres_users = local.db_engine != "POSTGRES" ? {} : { for user in var.users : user.name => user }
+  postgres_admins  = local.db_engine != "POSTGRES" ? {} : { for user in var.users : user.name => user if user.admin}
+  postgres_users = local.db_engine != "POSTGRES" ? {} : { for user in var.users : user.name => defaults(user, { db_ro = "", db_rw = ""}) if user.admin == false}
```

Are `db_ro` and `db_rw` not supposed to be lists?
```suggestion
  postgres_users = local.db_engine != "POSTGRES" ? {} : { for user in var.users : user.name => defaults(user, { db_ro = [], db_rw = []}) if user.admin == false}
```

---

You'd think so, yes. But, no, `terraform` doesn't accept that.
```
│ Invalid value for "defaults" parameter: .db_ro: invalid default value for string: string required.
```

---

Ah, right. That's where that problem is. Maybe `list(string)`?

---

That'd make those parameters required and indeed avoid the whole `null` issue. But at the cost at a more verbose configuration. I *might* go that route after splitting off the `admin` stuff from the *regular* users.